pkg/asset/asset.go

```diff
@@ -81,3 +86,105 @@ func AncestryPath(ancestors []string) string {
 	}
 	return strings.Join(revAncestors, "/")
 }
+
+// UnwrapCAIResource will unwrap a K8S resource from the CAI payload and populate any omitted fields.
+func UnwrapCAIResource(asset map[string]interface{}) (*unstructured.Unstructured, error) {
+	groupKind, found, err := unstructured.NestedString(asset, "asset_type")
+	if !found {
+		return nil, errors.Errorf("asset_type field not found")
+	}
+	if err != nil {
```

Comment:
should we check if err != nil first? otherwise we might get an error and found could default to false.

Reply:
fixed